File: users-api/src/main/java/net/plshark/users/auth/model/UserAuthSettings.java
package net.plshark.users.auth.model;

import javax.annotation.Nullable;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
import lombok.Builder;
import lombok.Value;

@Value
@Builder(toBuilder = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
@JsonDeserialize(builder = UserAuthSettings.UserAuthSettingsBuilder.class)
public class UserAuthSettings {

    @Nullable
    private final Long id;
    @Nullable
    private final Long userId;
    @JsonProperty("refresh_token_enabled") @Builder.Default
    private final boolean refreshTokenEnabled = true;

    @JsonPOJOBuilder(withPrefix = "")
    @JsonIgnoreProperties(ignoreUnknown = true)
    public static class UserAuthSettingsBuilder {

    }
}

File: user-error-api/src/main/kotlin/net/plshark/usererror/user/UserAuthSettings.kt
package net.plshark.usererror.user

import com.fasterxml.jackson.annotation.JsonProperty

data class UserAuthSettings(
    val id: Long?,
    @JsonProperty("user_id") val userId: Long?,
    @JsonProperty("refresh_token_enabled") val refreshTokenEnabled: Boolean = true,
    /** auth token expiration time in milliseconds */
    @JsonProperty("auth_token_expiration") val authTokenExpiration: Long?,
    /** refresh token expiration time in milliseconds */
    @JsonProperty("refresh_token_expiration") val refreshTokenExpiration: Long?
)
